search_service/proxy/atlas.py

```diff
@@ -0,0 +1,213 @@
+# default search page size
+from flask import current_app as app
+from typing import List, Dict
+
+from search_service.models.search_result import SearchResult
+from search_service.models.table import Table
+from search_service.proxy import BaseProxy
+from search_service.proxy.statsd_utilities import timer_with_counter
+
+DEFAULT_PAGE_SIZE = 10
+LOGGING = logging.getLogger(__name__)
+
+
+class AtlasProxy(BaseProxy):
+    TABLE_ENTITY = app.config['ATLAS_TABLE_ENTITY']
+    DB_ATTRIBUTE = app.config['ATLAS_DB_ATTRIBUTE']
+    NAME_ATTRIBUTE = app.config['ATLAS_NAME_ATTRIBUTE']
+    ATTRS_KEY = 'attributes'
+    REL_ATTRS_KEY = 'relationshipAttributes'
+
+    """
+    AtlasSearch connection handler
+    """
+    atlas: Atlas
+
+    def __init__(self, *,
+                 host: str = None,
+                 index: str = None,
+                 user: str = '',
+                 password: str = '',
+                 page_size: int = DEFAULT_PAGE_SIZE) -> None:
+        self.atlas = Atlas(host, username=user, password=password)
+        self.index = index
+        self.page_size = page_size
+
+    @staticmethod
+    def _entities(collections: EntityCollection) -> List[Entity]:
+        """
+        Helper method for flattening all collections from {collections}
+        :return: list of all entities
+        """
+        entities: List[Entity] = []
+        for collection in collections:
+            entities.extend(collection.entities)
+        return entities
+
+    def _parse_results(self, response: EntityCollection) -> List[Table]:
+        """
+        based on an atlas {response} with table entities, we map the required information
+        :return: list of tables
+        """
+        table_results = []
+        ids = list()
+        for hit in response:
+            ids.append(hit.guid)
+        # receive all entities
+        entities = self._entities(self.atlas.entity_bulk(guid=ids))
+        db_ids = []
+        for entity in entities:
+            relations = entity.relationshipAttributes
+            database = relations.get(self.DB_ATTRIBUTE)
+            if database:
+                db_ids.append(database['guid'])
+
+        # request databases
+        dbs_list = self._entities(self.atlas.entity_bulk(guid=db_ids)) if len(db_ids) > 0 else []
+        dbs_dict: Dict[str, Entity] = {db.guid: db for db in dbs_list}
+        for entity in entities:
+            relations = entity.relationshipAttributes
+            attrs = entity.attributes
+            database = relations.get(self.DB_ATTRIBUTE)
+            if database and database['guid'] in dbs_dict:
+                db_entity = dbs_dict[database['guid']]
+                db_attrs = db_entity.attributes
+
+                db_name = db_attrs.get(self.NAME_ATTRIBUTE)
+                db_qualified_name = db_attrs.get('qualifiedName', '').split('@')
+                db_cluster = db_qualified_name[1] if len(db_qualified_name) > 1 else ''
```

Comment:
Database entity has an attribute `clusterName`, which can be used directly here.